ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PlanNode.java

```diff
@@ -192,10 +195,13 @@ static void throwKeysNotIncludedError(
       final LogicalSchema schema
   ) {
     // When doing a `select *` key columns should be at the front of the column list
-    // but are added at the back during processing for performance reasons.
-    // Switch them around here:
-    final Stream<Column> keys = columns.stream()
-        .filter(c -> schema.isKeyColumn(c.name()));
+    // but are added at the back during processing for performance reasons. Furthermore,
+    // they should be selected in the order of the key, if that is not the same as the
+    // ordering within the value. Switch them around here:
```

Comment:
`the ordering within the value` sound confusing.

Reply:
yeah I read that back and totally agree - I'll reword